news/_posts/performance/2021-01-27-PHP-scalability-and-filesystem.md

```diff
@@ -0,0 +1,231 @@
+
+So, in simplest terms, if I want more clients, or customers, or visitors, I need to add resources to my system. Which seems fair, does it not?
+
+Of course, talking about performance here, we could argue that tuning the system's configuration and its different parts should achieve the same result. Still, once the fine tuning part is done, we should add resources to the system all the same in order to handle more load, so let's play along.
+
+Let's review together the two main levers to enable scalability - and keep in mind those are not exclusive, you often mix them up in Real Life™️.
+
+## Vertical
+
+The simpliest and most efficient way to add some resources to your system if often to add some RAM, or CPU, or storage, or all of those at the same time.
+
+![VerticalScaling][VerticalScaling]
+_Not enough mana!_
+{: .text-center }
+
+In a modern world made of cloud computing instances, vertical scaling can be done in a few mouse clicks by upgrading the instance type and releasing new resources. It can be a bit more tricky if you manage the hardware yourself, but it's still feasible.
+
+Though, keep in mind some potential limitations:
+
+* It tends to be less and less true, but pricing is not linear and 64GB of RAM may not cost twice 32GB
+* There may come a point where you can NOT add more resources verticaly and you'll just hit the limit
+* If you're managing your own infrastructure, you already know it takes time to upgrade or replace instances (buying, receiving and installing new parts can take weeks)
+* If you're managing your own infrastructure, what will you do with the old hardware?
+* Finally, if you've not already mixed a bit of horizontal scaling, **your system will be unavailable during the upgrade**! 
+
+Nonetheless, vertical scaling remains the most efficient way to scale a system or an application, as very little resources are wasted (either in networking, or scheduling, processing, etc..).
+
+Last but not least, vertical scaling usually requires no configuration change (either at the system's or application's level), keeping complexity to its minimum.
+
+## Horizontal
+
+As you may have guessed, horizontal scaling is all about adding more _instances_ of your system or application, whichever infrastructure you are using (either some docker, virtual machines or physical servers, the concept is similar):
+
+![HorizontalScaling][HorizontalScaling]
+_Just keep swimming, just keep swimming!_
+{: .text-center }
+
+Still, depending on your ecosystem, it may require some more work, at least a way to load balance traffic between the instances.
+
+And probably a way to distribute the application's files, especially for application like PrestaShop.
+
+Hence, the most simple architecture schema you will come with looks a lot like this (either load balancing with your kubernetes service through an ingress or just setting up your F5 BigIP appliance):
```

Comment:
I have no idea what are an ingress, a F5 or a BigIP appliance 😄  can we provide helper hrefs?

Reply:
Added some links, though I'm not sure the name dropping was that relevant.
It was mostly there to illustrate the many ways to perform load balancing, I hope it won't add to the confusion ^^